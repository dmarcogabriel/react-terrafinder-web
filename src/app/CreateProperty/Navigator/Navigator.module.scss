@import 'styles';

.navigator {
  margin: 2rem 0;
  display: flex;
  flex-direction: column-reverse;
  justify-content: space-between;
  padding: 0 2rem;

  @media ($desktop) {
    flex-direction: row;
    padding: 0;
  }

  .backButton {
    text-align: center;
    color: $blue;
    cursor: pointer;
    margin: 1rem 0;

    @media ($desktop) {
      font-size: 1.3rem;
      margin: 0;
    }
  }

  .nextButton {
    margin: 1rem 0;
    text-align: center;
    background: $blue;

    @media ($desktop) {
      margin: 0;
    }
  }
}
