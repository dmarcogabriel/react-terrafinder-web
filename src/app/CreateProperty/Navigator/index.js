import React from 'react';
import Button from 'common/components/Button';
import classes from './Navigator.module.scss';

export default function Navigator({
  onBack,
  onNext,
  nextButtonText = 'Pr√≥ximo',
}) {
  return (
    <div className={classes.navigator}>
      <button
        data-testid="back"
        className={classes.backButton}
        type="button"
        onClick={onBack}
      >
        Voltar
      </button>

      <Button dataTestId="next" onClick={onNext} className={classes.nextButton}>
        {nextButtonText}
      </Button>
    </div>
  );
}
