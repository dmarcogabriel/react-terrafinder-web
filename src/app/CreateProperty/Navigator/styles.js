import styled, { css } from 'styled-components';
import Button from 'common/components/atm/Button';

export const Container = styled.div`
  margin: 2rem 0;
  display: flex;
  flex-direction: column-reverse;
  justify-content: space-between;
  padding: 0 2rem;

  ${({ theme: { breakpoint } }) => css`
    @media (${breakpoint.desktop}) {
      flex-direction: row;
      padding: 0;
    }
  `}
`;

const BaseButton = styled(Button)`
  margin: 1rem 0;
  text-align: center;

  ${({ theme: { breakpoint } }) => css`
    @media (${breakpoint.desktop}) {
      margin: 0;
    }
  `}
`;

export const BackButton = styled(BaseButton).attrs({
  modifiers: ['secondary'],
})`
  ${({ theme: { breakpoint } }) => css`
    @media (${breakpoint.desktop}) {
      font-size: 1.3rem;
    }
  `}
`;

export const NextButton = styled(BaseButton)`
  ${({ theme: { color } }) => css`
    background: ${color.blue.df};
    color: ${color.white.df};
  `}
`;
