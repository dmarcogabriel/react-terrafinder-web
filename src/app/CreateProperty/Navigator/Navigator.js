import React from 'react';
import { Button } from '@mui/material';
import { NavigationContainer } from './styles';

export const Navigator = ({
  onBack,
  backButtonText = 'Voltar',
  onNext,
  nextButtonText = 'PrÃ³ximo',
}) => (
  <NavigationContainer sx={{ px: { md: 0, xs: 2 }, mt: 2 }}>
    <Button sx={{ my: { xs: 1, md: 0 } }} data-testid="back" onClick={onBack}>
      {backButtonText}
    </Button>

    <Button
      sx={{ my: { xs: 1, md: 0 } }}
      variant="contained"
      data-testid="next"
      onClick={onNext}
    >
      {nextButtonText}
    </Button>
  </NavigationContainer>
);
