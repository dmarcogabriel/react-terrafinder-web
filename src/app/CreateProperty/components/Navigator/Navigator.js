import React from 'react';
import { Button } from '@mui/material';
import { NavigationContainer } from './styles';

export const Navigator = ({
  onBack,
  backButtonText = 'Voltar',
  onNext,
  nextButtonText = 'PrÃ³ximo',
}) => (
  <NavigationContainer sx={{ px: { md: 0, xs: 2 }, mt: 2 }}>
    <Button
      sx={{ my: { xs: 1, md: 0 } }}
      variant="contained"
      color="info"
      data-testid="next"
      onClick={onNext}
    >
      {nextButtonText}
    </Button>

    <Button
      color="info"
      sx={{ my: { xs: 1, md: 0 } }}
      data-testid="back"
      onClick={onBack}
    >
      {backButtonText}
    </Button>
  </NavigationContainer>
);
