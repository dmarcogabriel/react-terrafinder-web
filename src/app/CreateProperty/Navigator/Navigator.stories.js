import React from 'react';
import { Navigator } from './Navigator';

const Template = (args) => <Navigator {...args} />;

export default {
  title: 'Create Property > Navigator',
  component: Navigator,
  argTypes: {},
};

export const NavigatorComponent = Template.bind();
NavigatorComponent.args = {
  nextButtonText: 'Pr√≥ximo',
};
