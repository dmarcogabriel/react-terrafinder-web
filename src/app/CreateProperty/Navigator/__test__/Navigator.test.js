import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import Navigator from '..';

describe('<Navigator />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Navigator />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Add tests here...

  it('matches snapshot', () => {
    const tree = renderer.create(<Navigator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
