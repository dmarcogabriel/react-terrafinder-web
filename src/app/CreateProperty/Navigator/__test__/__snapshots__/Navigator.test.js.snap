// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/CreateProperty/Navigator matches snapshot 1`] = `
.c1 {
  cursor: pointer;
  text-align: center;
  font-family: Roboto;
  box-shadow: 0 0.2rem 0.6rem 0 rgba(0,0,0,0.7);
  padding: 0.75rem;
  border-radius: 0.25rem;
  background: #fff;
  color: #16a5df;
}

.c3 {
  cursor: pointer;
  text-align: center;
  font-family: Roboto;
  box-shadow: 0 0.2rem 0.6rem 0 rgba(0,0,0,0.7);
  padding: 0.75rem;
  border-radius: 0.25rem;
  background: #fff;
  color: #333;
}

.c0 {
  margin: 2rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 2rem;
}

.c2 {
  margin: 1rem 0;
  text-align: center;
}

.c4 {
  background: #16a5df;
  color: #fff;
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 0;
  }
}

@media (min-width:768px) {
  .c2 {
    margin: 0;
    padding: 0.75rem 3rem;
  }
}

<div
  className="c0"
>
  <button
    className="c1 c2"
    data-testid="back"
    type="button"
  >
    Voltar
  </button>
  <button
    className="c3 c2 c4"
    data-testid="next"
    type="button"
  >
    Pr√≥ximo
  </button>
</div>
`;
